syft/frameworks/torch/tensors/interpreters/polynomial.py

```diff
@@ -1,257 +1,346 @@
+            self.exp_coeffs = torch.tensor(self.func_approx["exp"].coef)
+            self.tanh_coeffs = torch.tensor(self.func_approx["tanh"].coef)
+
+    def get_encrypt_function(self):
+        """The method encrypts the coefficients as required by the child tensor"""
+
+        if isinstance(self.child, FixedPrecisionTensor):
+
+            self.encrypt_fn["exp"] = getattr(torch.tensor(self.exp_coeffs), "fix_precision")()
+            self.encrypt_fn["sigmoid"] = getattr(
+                torch.tensor(self.sigmoid_coeffs), "fix_precision"
+            )()
+            self.encrypt_fn["tanh"] = getattr(torch.tensor(self.tanh_coeffs), "fix_precision")()
+
+        else:
+
+            self.encrypt_fn["exp"] = torch.tensor(self.exp_coeffs)
+            self.encrypt_fn["sigmoid"] = torch.tensor(self.sigmoid_coeffs)
+            self.encrypt_fn["tanh"] = torch.tensor(self.tanh_coeffs)
+
+    def add_function(self, name, function, degree=10, min_val=-10, max_val=10, steps=100):
+        """Add function to function_attr dictionary.
+
+           Args:
+             name[str]: Name of function
+             degree[int]: Degree of function
+             piecewise[List]: List of piecewise functions in format [min_val of fit,max_val of fit,step of fit,function to fit values]
+             function[callable]: Base function
+           """
+
+        self.function_attr[name + "_degree"] = degree
+        self.function_attr[name + "_function"] = function
+        self.func_approx[name] = self.interpolate(
+            function, [min_val, max_val, steps], degree=degree
+        )
+
+    def interpolate(
+        self, function: Callable, interval: List[Union[int, float]], degree: int = 10
+    ) -> np.poly1d:
+
+        """Returns a interpolated version of given function using Numpy's polyfit method.
+        
+           Ref: https://mortendahl.github.io/2017/04/17/private-deep-learning-with-mpc/
+
+         Args:
+             function (a lambda function): Base function to be approximated
+             interval (list of floats/integers): Interval of values to be approximated
+             degree (Integer): Degree of polynomial approximation
+             precision (Integer): Precision of coefficients
+
+         Returns:
+             f_interpolated (Numpy poly1d): Approximated Function
+         """
+
+        # function we wish to approximate
+        f_real = function
+        # interval over which we wish to optimize
+        f_interval = np.linspace(interval[0], interval[1], interval[2])
+
+        # interpolate polynomial of given max degree
+        coefs = np.polyfit(f_interval, f_real(f_interval), degree)
+
+        # reduce precision of interpolated coefficients
+        precision = self.precision
+        coefs = [int(x * 10 ** precision) / 10 ** precision for x in coefs]
+
+        # approximation function
+        f_interpolated = np.poly1d(coefs)
+
+        return f_interpolated
+
+    def sigmoid(self):
```

Comment:
this method is running sigmoid inline. This is incorrect given that sigmoid() does not have an underscore (also this means it will not work with autograd). It should be returning a new tensor with the results. self.child should be unchanged. 

If the other methods are this way as well they will also need to be modified.